Jenkinsfile

```diff
@@ -72,39 +71,66 @@ spec:
+                if (sh(script: "git checkout ${GIT_BRANCH_NAME}", returnStatus: true) == 0) {
+                    container('docker-cmds') {
+                        configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
+                            sh 'mvn -s $MAVEN_SETTINGS install -P quick'
+                        }
+                    }
+                }
+            }
+            dir ('Palisade-readers') {
+                git url: 'https://github.com/gchq/Palisade-readers.git'
+                if (sh(script: "git checkout ${GIT_BRANCH_NAME}", returnStatus: true) == 0) {
+                    container('docker-cmds') {
+                        configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
+                            sh 'mvn -s $MAVEN_SETTINGS install -P quick'
+                        }
+                    }
+                }
+            }
         }
-        stage('Unit Tests, Checkstyle and Install') {
-            //Repositories must get built in their own directory, they can be 'cd' back into later on
+
+        stage('Install, Unit Tests, Checkstyle') {
             dir ('Palisade-services') {
                 git url: 'https://github.com/gchq/Palisade-services.git'
-                sh "git fetch origin develop"
-                // CHANGE_BRANCH will be null unless you are building a PR, in which case it'll become your original branch name, i.e pal-xxx
-                // If CHANGE_BRANCH is null, git will then try to build BRANCH_NAME which is pal-xxx, and if the branch doesnt exist it will default back to develop
-                sh "git checkout ${env.CHANGE_BRANCH} || git checkout ${env.BRANCH_NAME} || git checkout develop"
+                sh "git checkout ${GIT_BRANCH_NAME}"
                 container('docker-cmds') {
                     configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
                         sh 'mvn -s $MAVEN_SETTINGS install'
                     }
                 }
             }
         }
+
         stage('Integration Tests') {
             dir ('Palisade-integration-tests') {
-            git url: 'https://github.com/gchq/Palisade-integration-tests.git'
-            sh "git fetch origin develop"
-            // CHANGE_BRANCH will be null unless you are building a PR, in which case it'll become your original branch name, i.e pal-xxx
-            // If CHANGE_BRANCH is null, git will then try to build BRANCH_NAME which is pal-xxx, and if the branch doesnt exist it will default back to develop
-            sh "git checkout ${env.CHANGE_BRANCH} || git checkout ${env.BRANCH_NAME} || git checkout develop"
+                git url: 'https://github.com/gchq/Palisade-integration-tests.git'
+                sh "git checkout ${GIT_BRANCH_NAME} || git checkout develop"
+                echo sh(script: 'git rev-parse --abbrev-ref HEAD', returnStdout: true)
```

Comment:
why is this needed?

Reply:
specifically the git rev parse

Reply:
`git reverse-parse --abbreviated-reference HEAD -> PAL-xxx-branch-name`
@nw1984 suggested printing out exactly which branch is being used, just for clarity (see above comment)

Reply:
when the sh git checkout line runs it will output "checked out x branch" 
so there will be logs available. we either need to do it for all git checkouts or none

Reply:
Yeah, i was just looking at jenkins, the branch checked out does get printed anyway

Reply:
for example, pulling from pal-515 jenkinsfile logs you get:

```
+ git checkout pal-515-JVM-Smoke-Tests
Branch 'pal-515-JVM-Smoke-Tests' set up to track remote branch 'pal-515-JVM-Smoke-Tests' from 'origin'.
Switched to a new branch 'pal-515-JVM-Smoke-Tests'
```